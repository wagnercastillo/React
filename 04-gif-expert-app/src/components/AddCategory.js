import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategorias }) => {
	const [inputValue, setInputValue] = useState("");

	const handleInputChange = (e) => {
		setInputValue(e.target.value);
	};

	const handleSubmit = (e) => {
		e.preventDefault();
		
		if (inputValue.trim().length > 2) {
			setCategorias((categorias) => [inputValue, ...categorias]);
			setInputValue('');
		}
	};
	return (
		<form onSubmit={handleSubmit}>
			<p>{inputValue}</p>
			<input 
				type="text"
				value={inputValue}
				onChange={handleInputChange} 
			/>
		</form>
	);
};

AddCategory.defaultProps = {
	setCategorias: { setCategorias: ["One Punch", "Samurai X", "Dragon Ball"] },
};

AddCategory.propTypes = {
	setCategorias: PropTypes.func.isRequired,
};
