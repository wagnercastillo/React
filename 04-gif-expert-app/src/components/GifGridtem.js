import React from "react";
import PropTypes from "prop-types";

export const GifGridtem = ({ id, title, url }) => {
	return (
		<div className="card animate__animated animate__bounce animate__delay-2s">
			<img src={url} alt={title} />
			<p>{title}</p>
		</div>
	);
};

GifGridtem.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
};
