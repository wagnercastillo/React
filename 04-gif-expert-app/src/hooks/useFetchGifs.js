import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (categoria) => {
	const [state, setState] = useState({
		data: [],
		loading: true,
	});

	useEffect(() => {
		getGifs(categoria).then((img) => {
			setState({
				data: img,
				loading: false,
			});
		});
	}, [categoria]);
	return state;
};
