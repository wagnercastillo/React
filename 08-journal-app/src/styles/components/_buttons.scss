.btn {
    background-color: transparent;
    border: none;
    color: white;
    cursor: pointer;
    font-size: 12px;
    padding: 7px 10px 7px 10px;

    &:focus {
        outline: none;
    }

    &:hover {
        color: darken($color: $primary, $amount: 15);
    }
}

.btn-primary {
    background-color: $primary;
    border-radius: 2px;


    &:disabled {
        background-color: lighten($color: $primary, $amount: 15);
    }

}

.btn-block{
    width: 100%;
}

.google-btn {
    cursor: pointer;
    margin-top: 5px;
    background-color: #4285f4;
    box-shadow: 0 3px 4px 0 rgba(0, 0, 0, 0.25);
    transition: box-shadow .3s ease;
    height: 42px;
    display: flex;
    align-items: center;
    justify-content: space-between;
}

.google-icon-wrapper {
    background-color: $white;
    height: 42px;
    width: 42px;
    display: flex;
    align-items: center;
    justify-content: center;
}

.google-icon {
    width: 22px;
}

.btn-text {
    padding: 0 14px;
    color: $white;
    font-size: 14px;
    letter-spacing: 0.2px;

    &:hover {
        box-shadow: 0 0 6px $google-blue
    }

    &:active {
        background: $button-active-blue
    }
}