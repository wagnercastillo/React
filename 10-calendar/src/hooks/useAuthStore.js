import { useDispatch, useSelector } from "react-redux"
import { calendarApi }  from "../api";
import { clearErrorMessage, onChecking, onLogin, onLogout } from "../store";

export const useAuthStore = () => {

    const { status, user, errorMessage } = useSelector( state => state.auth )
    const dispatch = useDispatch();

    const startLogin = async({email, password}) => {
        
        dispatch(onChecking());

        try {
            
            const { data } = await calendarApi.post('/auth', { email, password })

            localStorage.setItem('token', data.token );
            localStorage.setItem('token-init-date', new Date().getTime() );
            
            dispatch( onLogin({ name: data.name, uid: data.uid }) )
            


        } catch (error) {

            dispatch(onLogout('Credenciales Incorrectos'))
            
            setTimeout(() => {
                dispatch( clearErrorMessage() );
            }, 10);
        }
        
    }
    
    const startRegister = async({name, email, password}) => {

        dispatch(onChecking());
        try {

            const { data } = await calendarApi.post('/auth/new', { name, email, password })     

            localStorage.setItem('token', data.token );
            localStorage.setItem('token-init-date', new Date().getTime() );
            
            dispatch( onLogin({ name: data.name, uid: data.uid }) )


        } catch (error) {
            console.log(error);
            const errors = (
                error.response.data.errors?.password.msg || 
                error.response.data.errors?.name.msg || 
                error.response.data.errors?.email.msg ||
                error.response.data?.msg || '')

            dispatch(onLogout(errors))
            
            setTimeout(() => {
                dispatch( clearErrorMessage() );
            }, 10);
        }   

    }

    const checkAuthToken = async () => {

        const token = localStorage.getItem('token');
        if( !token ) return dispatch( onLogout() )

        try {

            const { data } = await calendarApi.get('/auth/renew');
            localStorage.setItem('token', data.token );
            localStorage.setItem('token-init-date', new Date().getTime() );
            dispatch( onLogin({ name: data.name, uid: data.uid }) )
            
        } catch (error) {
            localStorage.clear();
            dispatch( onLogout() )
        }
    }


    const startLogout = () => {
        localStorage.clear();
        dispatch(onLogout())
    }

    return {
        //* Propiedades
        errorMessage,
        status,
        user, 

        //* Metodos
        checkAuthToken,
        startLogin,
        startLogout,
        startRegister


    }
    
}

