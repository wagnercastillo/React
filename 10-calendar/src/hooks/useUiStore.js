import { useDispatch, useSelector } from "react-redux";
import { onOpenDateModal, onCloseDateModal } from '../store'

export const useUiStore = () => {

    const dispatch = useDispatch();

    const {
        isDateModalOpen
    } = useSelector(state => state.ui);

    const openDateModal = () => {
        dispatch(onOpenDateModal())
    }

    const closeDateModal = () => {
        dispatch(onCloseDateModal())
    }

    const toggleDateModal = () => {
        (isDateModalOpen)
            ? openDateModal()
            : closeDateModal();
    }

    return {
        // * Propiedades
        isDateModalOpen,

        // * Metodos
        openDateModal,
        closeDateModal,
        toggleDateModal,
    }

}