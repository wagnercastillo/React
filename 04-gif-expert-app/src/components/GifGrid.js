import React from "react";
import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridtem } from "./GifGridtem";

export const GifGrid = ({ categoria }) => {
	const { data: images, loading } = useFetchGifs(categoria);

	return (
		<>
			<h3 className="animate__animated animate_fadeIn">{categoria}</h3>

			{loading && <p className="animate__animated animate__fadeIn">Loading</p>}

			<div className="card-grid">
				{images.map((img) => (
					<GifGridtem key={img.id} {...img} />
				))}
			</div>
		</>
	);			
};

GifGrid.propTypes = {

	categoria: PropTypes.string.isRequired, 

}