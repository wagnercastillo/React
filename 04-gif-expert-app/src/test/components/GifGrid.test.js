import React from "react";
import "@testing-library/jest-dom";
import shallow from "enzyme/shallow";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en el componente <GifGrid />", () => {
	const categoria = "One Punch";

	test("debe de mostrar el componente correctamente", () => {
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true,
		});

		const wrapper = shallow(<GifGrid categoria={categoria} />);
		expect(wrapper).toMatchSnapshot();
	});

	test("debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs", () => {
		const gifs = [
			{
				id: "ABC",
				url: "https://localhost/cualquiercosa.jpg",
				title: "Cualquier cosa",
			},
			{
				id: "123",
				url: "https://localhost/cualquiercosa.jpg",
				title: "Cualquier cosa",
			},
		];

		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false,
		});

		const wrapper = shallow(<GifGrid categoria={categoria} />);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find("p").exists()).toBe(false);
		expect(wrapper.find("GifGridtem").length).toBe(gifs.length);
	});
});
