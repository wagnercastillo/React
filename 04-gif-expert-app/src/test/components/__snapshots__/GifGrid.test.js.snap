// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate_fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__fadeIn"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el componente <GifGrid /> debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate_fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridtem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa.jpg"
    />
    <GifGridtem
      id="123"
      key="123"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa.jpg"
    />
  </div>
</Fragment>
`;
